### PR TITLE
feat: added onCompleted support

### PR DESCRIPTION
**Pull request by tea-lover-418**
We added onCompleted support as this was a functionality every user probably implements on their own in a similar way. Let me know if you need any changes / or would like a different approach.

@ugogo

### CODE REVIEW

File: src/index.tsx [hunk 1]
```diff
@@ -11,14 +11,16 @@ const KEY_CODE = {
...
+  onChange?: (data: string) => any;
+  onCompleted?: (data: string) => any;
   placeholder?: string;
   value?: string;
 };
 
 const ReactInputVerificationCode = ({
   length = 4,
   onChange,
+  onCompleted,
```
Comment (ugogo) [inline src/index.tsx:23]:
```suggestion
  onCompleted = () => {},
```

This will avoid having to check `onCompleted` type everytime

**Comment by ugogo** [comment]
Hey @tea-lover-418, thanks for the PR!

I only have one suggestion, otherwise, it looks good to me!

File: src/index.tsx (hunk 1 above)
Comment (stijn-gravity) [reply src/index.tsx]: Alright, I did the same for onChange aswell.

Comment (tea-lover-418) [reply src/index.tsx]: Alright, I did the same for onChange aswell

File: src/index.tsx [hunk 2]
```diff
@@ -137,7 +139,13 @@ const ReactInputVerificationCode = ({
...
 
   React.useEffect(() => {
-    onChange(value.join(''));
+    const stringValue = value.join('')
+
+    onChange(stringValue);
+
+    if (!stringValue.includes(placeholder)) {
+      onCompleted(stringValue)
+    }
```
Comment (ugogo) [inline src/index.tsx:148]:
```suggestion
    const stringValue = value.join('');
    const isCompleted = !stringValue.includes(placeholder);

    if (isCompleted) onCompleted(stringValue);
    onChange(stringValue);
```

A little bit of code cleaning to match the current codebase and we are good to go 🙏

**Comment by ugogo** [comment]
Thanks for your contribution @tea-lover-418!

**Pull request merged**